### ISSUE TITLE
Loss of Tacotron

### ISSUE THREAD

**Issue by chankl3579** [CorentinJ/Real-Time-Voice-Cloning#725]
May I ask what actually is the loss while training the synthesizers. 

For the new Pytorch repo, as mentioned in #653 , the loss is the sum of:
1. L1 loss +L2 loss of decoder output
2. L2 loss of Mel spectrogram after Post-Net
3. Cross entropy of Stop Token

I can also see it in the code: 
` # Backward pass `
` m1_loss = F.mse_loss(m1_hat, mels) + F.l1_loss(m1_hat, mels) `
` m2_loss = F.mse_loss(m2_hat, mels) `
` stop_loss = F.binary_cross_entropy(stop_pred, stop) `
` loss = m1_loss + m2_loss + stop_loss `

However, what is the loss in the old Tensorflow repo? 
In the original paper it simply mentioned 
"We extend [15] by augmenting the L2 loss on the predicted spectrogram with an additional L1 loss. "
The RTVC thesis it is stated that 
"The loss function is the L2 loss between the predicted and ground truth mel spectrograms. "

In the code there are some items related to loss including
eval_losses, before_losses, after_losses, stop_token_losses, linear_losses, linear_loss
Are they related to the loss? Or did I miss the lines about the loss?

**Comment by blue-fish** [comment]
It is documented here: https://github.com/Rayhane-mamah/Tacotron-2/wiki/Spectrogram-Feature-prediction-network#training

Our old Tensorflow code in this repo modified the Rayhane-mamah code to add L1 loss of decoder output. That code is obsolete and no longer supported. Please file an issue in https://github.com/Rayhane-mamah/Tacotron-2 if you have more questions about the Tensorflow synthesizer.

**Comment by chankl3579** [comment]
Thanks for the reference link !!
